### PR TITLE
Make naming of 'getXxxById' consistent across the project

### PR DESCRIPTION
**Pull request by barroij**
The majority of occurrence is `getXxxByID` but there are a few `getXxxById`. This PR purpose is to always use `getXxxByID`

That's not a big deal and it's breaking backward compatibility, so it might not be worth it. But if it has to be fixed, a major version like 5.0 might a the good time to do it.

So it's up to you guys @deltakosh @RaananW & Co :)

Just close it if you don't want of it

### CODE REVIEW

**Comment by RaananW** [comment]
Well, that makes total sense :-)

I am actually more for `getXXXById` and less for `byID`. especially since the browser uses `Id` and not `ID` (i.e. `getElementById`). And it is possible to have a proxy to old functions, something along the lines of:

```javascript
public get getXXXByID() {
  return this.getXXXById;
}
```

This way we keep consistency, backwards compatibility, and standards :-)

**Comment by Popov72** [comment]
And you can tag the old ones with something like `/** @deprecated Please use getXXXById instead */`

**Comment by barroij** [comment]
@RaananW @Popov72 
I made the changes almost everywhere.

These ones are let unchanged : 
`Ligth.getTypeID`
`SpriteMap.getTileID` 
`InstancesBlock.instanceID`
`ICustomAnimationFrameRequester.requestID`

When the change was a public property, I changed them and added a deprecated get/set proxy (even for hidden props).

File: gui/src/2D/xmlLoader.ts [hunk 1]
```diff
@@ -289,6 +289,11 @@ export class XmlLoader {
         return this._isLoaded;
     }
 
+    /** @deprecated Please use getNodeById instead */
+     public getNodeByID(id: string): any {
```
Comment (RaananW) [inline gui/src/2D/xmlLoader.ts:293]: Is that needed?

File: src/PostProcesses/postProcess.ts [hunk 2]
```diff
@@ -884,7 +884,7 @@ export class PostProcess {
             return null;
         }
 
-        var camera = scene ? scene.getCameraByID(parsedPostProcess.cameraId) : null;
+        var camera = scene ? scene.getCameraByI(parsedPostProcess.cameraId) : null;
```
Comment (RaananW) [inline src/PostProcesses/postProcess.ts:887]: missing a `d`

File: src/XR/features/WebXRImageTracking.ts [hunk 3]
```diff
@@ -150,6 +150,11 @@ export class WebXRImageTracking extends WebXRAbstractFeature {
         return typeof XRImageTrackingResult !== "undefined";
     }
 
+    /** @deprecated Please use getTrackedImageById instead */
+    public getTrackedImageByID(id: number): Nullable<IWebXRTrackedImage> {
```
Comment (RaananW) [inline src/XR/features/WebXRImageTracking.ts:154]: probably not needed?

File: src/scene.ts [hunk 4]
```diff
@@ -2737,12 +2757,17 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
...
+        return this.getCameraByI(id);
+    }
+
     /**
      * Gets a camera using its id
      * @param id defines the id to look for
      * @returns the camera or null if not found
      */
-    public getCameraByID(id: string): Nullable<Camera> {
+    public getCameraByI(id: string): Nullable<Camera> {
```
Comment (RaananW) [inline src/scene.ts:2770]: missing a `d`

File: src/scene.ts [hunk 5]
```diff
@@ -2752,12 +2777,17 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
...
+        return this.getCameraByUniqueI(uniqueId);
+    }
+
     /**
      * Gets a camera using its unique id
      * @param uniqueId defines the unique id to look for
      * @returns the camera or null if not found
      */
-    public getCameraByUniqueID(uniqueId: number): Nullable<Camera> {
+    public getCameraByUniqueI(uniqueId: number): Nullable<Camera> {
```
Comment (RaananW) [inline src/scene.ts:2790]: missing a `d`

**Review by RaananW — CHANGES_REQUESTED** [review]
Thanks! just a few small changes

File: gui/src/2D/xmlLoader.ts (hunk 1 above)
Comment (barroij) [reply gui/src/2D/xmlLoader.ts:293]: you tell me :) I can remove it

**Comment by barroij** [comment]
@RaananW I fixed the missing  `d`s and removed the changes that were not needed  (xmlLoader.ts & ebXRImageTracking.ts )

**Comment by RaananW** [comment]
> @RaananW I fixed the missing `d`s and removed the changes that were not needed (xmlLoader.ts & WebXRImageTracking.ts )

I might have missed one - are the ones in viewerManager required?

File: gui/src/2D/xmlLoader.ts (hunk 1 above)
Comment (RaananW) [reply gui/src/2D/xmlLoader.ts]: you just added it, no? there is no need to add new functions, just change existing ones

Comment (barroij) [reply gui/src/2D/xmlLoader.ts]: yes you're right, I put myself in automatic mode and add a few unwanted deprecated function 👎 .

**Comment by barroij** [comment]
@RaananW it should be ok now, I removed the ones in viewerManager. sorry for that 😕

**Review by RaananW — APPROVED** [review]

File: src/Behaviors/Meshes/pointerDragBehavior.ts [hunk 6]
```diff
@@ -34,10 +34,18 @@ export class PointerDragBehavior implements Behavior<AbstractMesh> {
      * @hidden
      */
     public _useAlternatePickedPointAboveMaxDragAngle = false;
+    /** @deprecated Please use currentDraggingPointerId instead */
+    public get currentDraggingPointerID(): number {
```
Comment (RaananW) [inline src/Behaviors/Meshes/pointerDragBehavior.ts:38]: Are these needed? I don't see the change of the existing function.

File: src/Behaviors/Meshes/pointerDragBehavior.ts [hunk 7]
```diff
@@ -34,10 +34,18 @@ export class PointerDragBehavior implements Behavior<AbstractMesh> {
...
+    }
+    /** @deprecated Please use currentDraggingPointerId instead */
+    public set currentDraggingPointerID(currentDraggingPointerID: number) {
+        this.currentDraggingPointerId = currentDraggingPointerID;
+    }
     /**
      * The id of the pointer that is currently interacting with the behavior (-1 when no pointer is active)
      */
-    public currentDraggingPointerID = -1;
+    public currentDraggingPointerId = -1;
```
Comment (RaananW) [inline src/Behaviors/Meshes/pointerDragBehavior.ts:48]: This changes back-compat, requires getter.

File: src/Behaviors/Meshes/sixDofDragBehavior.ts [hunk 8]
```diff
@@ -40,10 +40,18 @@ export class SixDofDragBehavior implements Behavior<Mesh> {
      * The distance towards the target drag position to move each frame. This can be useful to avoid jitter. Set this to 1 for no delay. (Default: 0.2)
      */
     public dragDeltaRatio = 0.2;
+    /** @deprecated Please use currentDraggingPointerId instead */
+    public get currentDraggingPointerID(): number {
```
Comment (RaananW) [inline src/Behaviors/Meshes/sixDofDragBehavior.ts:44]: Same - are these needed?

File: src/Behaviors/Meshes/sixDofDragBehavior.ts [hunk 9]
```diff
@@ -40,10 +40,18 @@ export class SixDofDragBehavior implements Behavior<Mesh> {
...
+    }
+    /** @deprecated Please use currentDraggingPointerId instead */
+    public set currentDraggingPointerID(currentDraggingPointerID: number) {
+        this.currentDraggingPointerId = currentDraggingPointerID;
+    }
     /**
      * The id of the pointer that is currently interacting with the behavior (-1 when no pointer is active)
      */
-    public currentDraggingPointerID = -1;
+    public currentDraggingPointerId = -1;
```
Comment (RaananW) [inline src/Behaviors/Meshes/sixDofDragBehavior.ts:54]: Same as above - back-compat

**Review by RaananW — CHANGES_REQUESTED** [review]
Sorry, went over it again and have a few changes.
It's a big semantic change, we want to be sure back-compat is guaranteed.

There is also a small doc issue

**Comment by barroij** [comment]
@RaananW I'm closing this one and make another branch in which I'll check the changes one by one.

**Pull request closed without merging**